setuptools_rust/setuptools_ext.py

```diff
@@ -17,6 +20,45 @@
 
 
 def add_rust_extension(dist):
+    sdist_base_class = dist.cmdclass.get("sdist", sdist)
+    sdist_options = sdist_base_class.user_options.copy()
+    sdist_boolean_options = sdist_base_class.boolean_options.copy()
+    sdist_negative_opt = sdist_base_class.negative_opt.copy()
+    sdist_options.extend([
+        ('vendor-crates', None,
+         "vendor Rust crates"),
+        ('no-vendor-crates', None,
+         "don't vendor Rust crates."
+         "[default; enable with --vendor-crates]"),
+    ])
+    sdist_boolean_options.append('vendor-crates')
+    sdist_negative_opt['no-vendor-crates'] = 'vendor-crates'
+
+    class sdist_rust_extension(sdist_base_class):
+        user_options = sdist_options
+        boolean_options = sdist_boolean_options
+        negative_opt = sdist_negative_opt
+
+        def initialize_options(self):
+            super().initialize_options()
+            self.vendor_crates = 0
+
+        def get_file_list(self):
+            if self.vendor_crates:
+                base_dir = self.distribution.get_fullname()
+                vendor_path = os.path.join(base_dir, "vendored-crates")
+                cargo_config = subprocess.check_output(["cargo", "vendor", vendor_path])
```

Comment:
Rust extensions may be in a subdirectory. For example PyCA cryptography has ``src/rust/Cargo.toml``. I think you need to iterate over all ``RustExtension`` instances and run ``["cargo", "vendor", "--manifest-path", rustext.path, vendor_path]``. For projects with more than one ``RustExtension`` you can use the ``--sync`` option.

Reply:
Thanks for the advice. I'll try to address it later.